apps/infra/analytics/contract_analytics.py

```
@@ -0,0 +1,785 @@
+            )
+            .group_by("user_type")
+            .agg(pl.count().alias("user_count"))
+        )
+
+        return user_segments.to_dicts() if user_segments.shape[0] > 0 else []
+    except Exception as e:
+        logger.error(f"Error analyzing user segments: {e}")
+        return []
+
+
+def get_contract_summary(catalog, chain_id, contract_address, time_window=None):
```

Pylint (3.3.7)

[refactor] Too many local variables (20/15)

(R0914)